src/components/expression/expression.js

```diff
@@ -79,17 +80,25 @@ genotet.ExpressionView = function(viewName, params) {
 
   // Set up data loading callbacks.
   $(this.container).on('genotet.ready', function() {
+    this.data.tfaFileName = params.tfaFileName;
```

Comment:
I don't get it. You are requesting a file from the client. The client must know the filename already. Letting the server return filename is redundant.